app/views/links/edit.html.erb

```diff
@@ -6,22 +6,33 @@
   <h2><%= @interaction.label %></h2>
 </div>
 
-<% if flash[:bad_url] %>
-<div class="alert alert-danger">Please enter a valid link.</div>
+<% if flash[:failed_action] %>
+<div class="alert alert-danger"><%= flash[:failed_action] %></div>
 <% end %>
 
 <p><b>LGSL</b> <%= @service.lgsl_code %><b class="add-left-margin">LGIL</b> <%= @interaction.lgil_code %></p>
 
-<%= label_tag 'link', 'URL' %>
+<%= form_for @link, url: local_authority_service_interaction_links_path(@local_authority.slug, @service.slug, @interaction.slug), method: "put" do |f| %>
+  <div class="form-group <% if flash[:failed_action] %>has-error<% end %>">
+    <div class='col-xs-12 no-gutter'>
+      <%= label_tag 'link', 'URL' %>
```

Comment:
The label we generate here looks like `<label for="link">URL</label>` which means it's the label for an input with the id `link`, but the input we generate has the id `link_url`.  If we replace this call with `<%= f.label :url, 'URL' %> then it'll do the right thing.  In this version we ask the form builder to generate the label rather than rendering it ourselves (much like we swapped from `text_field_tag` to `f.text_field` to render the input field) and so it joins things up properly for us.  See: http://api.rubyonrails.org/classes/ActionView/Helpers/FormHelper.html#method-i-label for more info on the `f.label` helper specificially and that page for most of the form builder helpers you can call on `f` in general.